views/cypress/utils/set-response.js

```diff
@@ -116,3 +118,111 @@ export function setResponse(widgetSelector, widgetActiveSelector, qtiClass) {
 
     cy.get(`${widgetActiveSelector} button.widget-ok`).click({ force: true });
 }
+
+function previewItem () {
+    cy.intercept('GET', urls.itemPreview).as('preview');
+    cy.get(selectors.previewItemButton).should('not.have.class', 'disabled');
+    cy.get(selectors.previewItemButton).click({force :true});
+    cy.wait('@preview');
+    cy.get('.qti-choiceInteraction').should('exist');
+}
+
+function saveItem() {
+    cy.intercept('POST', '**/saveItem*').as('saveItem');
+    cy.get('[data-testid="save-the-item"]').click();
+    cy.wait('@saveItem')
+        .its('response.body')
+        .its('success')
+        .should('eq', true);
+}
+
+export function addResponseProcessing(
+    widgetSelector,
+    qtiClass,
+    responseProcessingOptions
+) {
+    cy.log('ADD RESPONSE PROCESSING', widgetSelector, qtiClass);
+    cy.getSettled(widgetSelector).find( selectors.selectInteractionResponse).click({force: true});
+
+    switch (responseProcessingOptions) {
+        case 'match correct':
+            cy.get('ol').find('li[data-identifier="choice_3"]').click({force: true});
```

Comment:
I would have moved these three blocks of code to its own functions, so `addResponseProcessing` is more clean. But as seems they don't need to be reused it is also ok as it is.